File: 125.ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        int c1 = 0;
        int c2 = s.length() - 1;

        while(c1 < c2) {
            char ch1 = s.charAt(c1);
            char ch2 = s.charAt(c2);

            if(!Character.isLetter(ch1) && !Character.isDigit(ch1)) {
                c1++;
                continue;
            } 

            if(!Character.isLetter(ch2) && !Character.isDigit(ch2)) {
                c2--;
                continue;
            }

            if(ch1 != ch2) {
                return false;
            }
            c1++;
            c2--;
        }

        return true;
    }
}

File: 125.ValidPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        c1 = 0
        c2 = len(s) - 1

        while c1 < c2:
            ch1 = s[c1].lower()
            ch2 = s[c2].lower()
            if not ch1.isdigit() and not ch1.isalpha():
                c1+=1
                continue
            if not ch2.isdigit() and not ch2.isalpha():
                c2-=1
                continue

            if ch1 != ch2:
                return False

            c1+=1
            c2-=1
          
        return True
